File: server/datasets/controlled_benchmarks/python/multiple_files/multiple_b2/organic_2.py
# Two Sum (hash map) - organic submission 2
# Two-pass: build the full index map first, then scan for complements.

def two_sum(nums, target):
    index_map = {}
    for i, num in enumerate(nums):
        index_map[num] = i

    for i, num in enumerate(nums):
        complement = target - num
        if complement in index_map and index_map[complement] != i:
            return [i, index_map[complement]]
    return None


def main():
    nums = [3, 2, 4]
    target = 6
    print("Indices:", two_sum(nums, target))


if __name__ == "__main__":
    main()

File: server/datasets/controlled_benchmarks/java/multiple_files/multiple_b2/organic_2.java
// Two Sum (hash map) - organic submission 2
// Two-pass: build the full index map first, then scan for complements.

import java.util.Arrays;
import java.util.HashMap;
import java.util.Map;

class Solution {
    public static int[] twoSum(int[] nums, int target) {
        Map<Integer, Integer> indexMap = new HashMap<>();
        for (int i = 0; i < nums.length; i++) {
            indexMap.put(nums[i], i);
        }

        for (int i = 0; i < nums.length; i++) {
            int complement = target - nums[i];
            if (indexMap.containsKey(complement) && indexMap.get(complement) != i) {
                return new int[]{i, indexMap.get(complement)};
            }
        }
        return null;
    }

    public static void main(String[] args) {
        int[] nums = {3, 2, 4};
        int target = 6;
        System.out.println("Indices: " + Arrays.toString(twoSum(nums, target)));
    }
}
